### DISCUSSION TITLE
Cannot access Nuxt instance from inside Pinia's store

### DISCUSSION THREAD

**Discussion by josephlevarato** [nuxt/nuxt#16795] [Questions]
Hello,

I'm trying to setup a theme switcher system, but I'm having a hard time to store it in cookies.

I installed [nuxt-cookies](https://github.com/microcipcip/cookie-universal/tree/master/packages/cookie-universal-nuxt#readme) and added it in my `nuxt.config.ts`. I didn't use an alias, so I guess it's available with `this.$cookies`.

My store looks like so:

```ts

import { defineStore } from 'pinia'

const { $cookies } = useNuxtApp()

export const useThemeStore = defineStore('theme', () => {

    const current = ref(initTheme())
    function switchTheme (newTheme: string) {
        current.value = newTheme

        $cookies.set('ovrld-theme', newTheme, {
            path: '/',
            maxAge: 60 * 60 * 24 * 7
        })
    }

    return { current, switchTheme }
})

const initTheme = () => {
    const cookieTheme = $cookies.get('ovrld-theme')

    if (cookieTheme) {
        return cookieTheme.theme
    } else {
        $cookies.set('ovrld-theme', 'dark', {
            path: '/',
            maxAge: 60 * 60 * 24 * 7
        })

        return 'dark'
    }
}
```

However, the following line:

```ts
const { $cookies } = useNuxtApp()
```

Makes me have this error:

```
[nuxt] [request error] [unhandled] [500] nuxt instance unavailable
  at Module.useNuxtApp (./.nuxt/dist/server/server.mjs:485:13)  
  at $id_H94HSl2Hhb (./.nuxt/dist/server/server.mjs:3612:44)  
  at async __instantiateModule__ (./.nuxt/dist/server/server.mjs:6446:3)
```

I found a [discussion](https://github.com/nuxt/framework/discussions/5321) saying that it should work, but maybe my syntax isn't correct?

Thank you in advance

**Comment by sevillaarvin** [comment]
It says here that Nuxt 3 is not supported: https://github.com/microcipcip/cookie-universal/issues/133#issuecomment-1204513904

Perhaps you could use `useCookie` instead? https://v3.nuxtjs.org/api/composables/use-cookie#usecookie
